xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -389,6 +391,18 @@ func (edsImpl *edsBalancerImpl) handleSubConnStateChange(sc balancer.SubConn, s
 	}
 }
 
+// updateConfig handles changes to the circuit breaking configuration.
+func (edsImpl *edsBalancerImpl) updateConfig(edsConfig *EDSConfig) {
+	if edsImpl.counter == nil || edsImpl.counter.ServiceName != edsConfig.EDSServiceName {
+		edsImpl.counter = &client.ServiceRequestsCounter{ServiceName: edsConfig.EDSServiceName}
+	}
+	edsImpl.counter.UpdateService(edsConfig.CircuitBreaking, edsConfig.MaxRequests)
```

Comment:
You shouldn't update the service of this count. You should update the counter.

There should be a `map[string]counter` in the client package. And you set `edsImpl.counter` to the corresponding value. When service name changes, you read the new value, and set `edsImpl.counter`.

Reply:
Made this change.